### ISSUE TITLE
Duplicate entry with CardDAV discovery

### ISSUE THREAD

**Issue by army1349** [blind-coder/rcmcarddav#256]
With preset URL specified like:
```
$prefs['Cloud'] = array('
    'url' =>  'https://example.com/remote.php/dav/',
...
```
CardDAV discovery is is done correctly, first contact group is added successfully like Cloud (Phonebook 1), but DB insert fails for second contact group because of (user_id, presetname) unique index:
```
dbmigrations/0005-changemysqlut8toutf8mb4/mysql.sql:
ALTER TABLE TABLE_PREFIXcarddav_addressbooks DROP INDEX `user_id`, ADD UNIQUE `user_id` (`user_id`, `presetname`(191)) USING BTREE;
```

**Comment by blind-coder** [comment]
That happens when the server does not provide a displayname for the addressbook. In that case, you have to specify a seperate preset for each addressbook, or set displaynames on the CardDAV server.

**Comment by army1349** [comment]
If (user_id, presetname) has to be unique, then I don't see a way around except separate preset for each addressbook.
Does it need to be this way?

**Comment by blind-coder** [comment]
If you can set a displayname on your addressbook server, that would work as well.

**Comment by army1349** [comment]
How is that? presetname will be set according to displayname from addressbook server?

**Comment by blind-coder** [comment]
Correct.

**Comment by army1349** [comment]
I have tested it. Nextcloud CardDAV backend correctly returns displayname and yet presetname value in DB is set set to name of the preset from config for all addressbooks.

**Comment by blind-coder** [comment]
That's weird, I know that that worked fine before.
Can you post your confg.inc.php file?

**Comment by army1349** [comment]
Sure.
```
$prefs['Cloud'] = array(
        'name'         =>  'Cloud',
        'username'     =>  '%u',
        'password'     =>  '%p',
        'url'          =>  'https://cloud.example.com/remote.php/dav/',
        'active'       =>  true,
        'readonly'     =>  false,
        'refresh_time' => '02:00:00',
        'fixed'        =>  array( 'name', 'username', 'password' ),
        'hide'        =>  false,
);
```

**Comment by blind-coder** [comment]
I was able to reproduce this here now, and after investigating this, it seems the UNIQUE key on user_id,presetname is simply wrong. I have added a db migration to change that to an INDEX on user_id only in the branch issue_256. Could you test that?

**Comment by army1349** [comment]
Migration script works and I have used it with exactly same index as in this correction for some time and I didn't have any issues, so I guess, it should be OK. Thanks!

**Comment by blind-coder** [comment]
Thank you for the confirmation. I'll merge this into master and create a new release soon!

**Comment by mstilkerich** [comment]
This issue is fixed by commit ed36d7a1aad3f756149904b6a5101e80a14cbe3d. If you find the issue persists, please comment / reopen the issue.
